base/util.jl

```diff
@@ -564,7 +564,11 @@ function runtests(tests = ["all"]; ncores::Int = ceil(Int, Sys.CPU_THREADS::Int
     ENV2 = copy(ENV)
     ENV2["JULIA_CPU_THREADS"] = "$ncores"
     try
-        run(setenv(`$(julia_cmd()) $(joinpath(Sys.BINDIR::String,
+        cmd = Base.julia_cmd()
+        if Base.JLOptions().use_sysimage_native_code == 0
+            push!(cmd.exec, "--sysimage-native-code=no")
```

Comment:
IMO this should be a part of `Base.julia_cmd()`.

Reply:
What do you mean?

Reply:
I mean that we should do this check inside of `Base.julia_cmd()` and push the arguments onto that `cmd` object, instead of only doing it here.

Reply:
https://github.com/JuliaLang/julia/pull/42185